Add property datasets

Jacob-Chmura opened the pull request:
Close #2

Jacob-Chmura committed:
Add template for weight dataset

Jacob-Chmura committed:
Update prompt

Jacob-Chmura committed:
Add data

Jacob-Chmura committed:
Fix markdown typo

jonahdvt committed:
Create datasets

jonahdvt committed:
Accidentally created dataset file

jonahdvt committed:
items data added

file with ~900 items that have been categorized in terms of phase, size, weight, sentience, rigidity (the ones from the paper except for mass-count)

jonahdvt committed:
Update data.csv

jonahdvt committed:
Delete data/items.xlsx

jonahdvt committed:
Update README.md on data collection

Jacob-Chmura commented on README.md:
Let's revert this change and put the prompting info directly into the datafile `README.md`

Jacob-Chmura commented on data/property_data/weight_data/data.csv:
Given that this has _all_ the properties, and not just the _weight_, we should rename the directories to reflect that

Jacob-Chmura committed:
Refactor README files

Jacob-Chmura commented on data/property_data/weight_data/README.md:
```diff
@@ -0,0 +1,24 @@
+1. Add another column that numerically categorizes each item according to it's physical phase with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's sentience with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's weight with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's rigidity with respect to the reference points: rock, tree, ant, cat, chimp, man.
```
@jonahdvt Please update the reference points for the _non-sentience_ categories in accordance with how you prompted the model.

Jacob-Chmura committed:
Remove 'weight_data' subdirectory

Jacob-Chmura committed:
Merge branch 'master' into data/property-data

Jacob-Chmura commented on data/property_data/data.csv:
```diff
@@ -0,0 +1,906 @@
+Moon,9,4,1,7,5
+Sun,9,4,1,7,5
+Cloud,9,1,1,7,1
+Rain,1,1,1,1,1
```
@jonahdvt Why are these numerically encoded as 9 when there are only 5 reference points?

Jacob-Chmura commented on data/property_data/data.csv:
### Distinct Items
@jonahdvt did you have a look at the data? I still see several duplicates. More importantly, you can see that, in some cases, the duplicate samples are classified differently:

![foo](https://github.com/user-attachments/assets/fc2ab5e9-f071-4ef6-adca-d3e87acbfff1)

jonahdvt replied:
we can remove the duplicates. if anything the closeness in classification shows confidence in the model. it only differs very slightly so it shows that the data is classified pretty well. would be more concerning if they were vastly different

Jacob-Chmura commented on data/property_data/data.csv:
### Data Distribution
It's unlikely that the model would have by happen-chance, generated items that uniformly cover our reference points. But in case you haven't looked at the distribution, here is what it looks like:

![foo](https://github.com/user-attachments/assets/d91a758a-159b-4b72-aed9-74acd1651ad0)

And perhaps the skew is more apparent in this chart:

![foo](https://github.com/user-attachments/assets/b36d2529-5cdd-42c2-bae1-d596d62b87e9)

### Conclusion
I suggest we hand prompt for some of the missing categories, or we re-think our reference points to make them a bit easier to discriminate. In either case, we must investigate duplicates and _out-of-domain_ values like 7-9.

jonahdvt committed:
Fixed reference points of prompts and added note

jonahdvt committed:
Update data.csv

with the new unlabelled set to unskew data. removed duplicates. still need to recaliber size and weight ranges

jonahdvt committed:
Reformat data.csv

Jacob-Chmura commented on data/property_data/README.md:
```diff
@@ -0,0 +1,25 @@
+</blockquote>
+
+_Note_: the LLM had to be repeatedly prompted until about 1000 distinct items were generated. To maintain attention, the property attributes were generated in baches of 100 items.
+_Note_: some indices in the size category were registered as "out of distribution", as there were points 1-9. That is due to the fact that size was too challenging for the LLM to categorise, hence the categories of "smaller than ant" and "larger than stadium" were added, and the other categories were made "in between" reference points. THis results in 9 instead of 8 categories.
```
I would prefer if we just remap them to a common range. With this explanation alone, it's not super clear why the LLM generated numbers beyond the prescribed bins. Maybe I'm missing something, but I feel that having a consistent range is better for downstream perf anyway

Jacob-Chmura commented:
@jonahdvt Let's try to precompute train/test data splits as seems to be standard when working with these hugging face libraries.

jonahdvt committed:
Update README.md for 3 extra categories

jonahdvt committed:
categorised new items and normalised 1-5

jonahdvt committed:
added last 2 categories

jonahdvt committed:
last 2 categories added and done. data done
